src/server/game/Globals/AreaTriggerDataStore.cpp

```diff
@@ -16,17 +16,48 @@
  */
 
 #include "AreaTriggerDataStore.h"
+#include "AreaTrigger.h"
 #include "AreaTriggerTemplate.h"
 #include "DatabaseEnv.h"
 #include "DB2Stores.h"
 #include "Log.h"
+#include "MapManager.h"
 #include "ObjectMgr.h"
 #include "Timer.h"
 #include <cmath>
 
+struct AreaTriggerTemplateKey
+{
+    uint32 Id;
+    bool IsServer;
+
+    AreaTriggerTemplateKey(uint32 id, bool isServer) : Id(id), IsServer(isServer) { }
+};
+
+bool operator ==(AreaTriggerTemplateKey const& A, AreaTriggerTemplateKey const& B)
+{
+    return A.Id == B.Id && A.IsServer == B.IsServer;
+}
+
+namespace std
+{
+    template <>
+    struct hash<AreaTriggerTemplateKey>
+    {
+        std::size_t operator()(AreaTriggerTemplateKey const& value) const
+        {
+            size_t hashVal = 0;
+            Trinity::hash_combine(hashVal, value.Id);
+            Trinity::hash_combine(hashVal, value.IsServer);
+            return hashVal;
+        }
+    };
+}
+
 namespace
 {
-    std::unordered_map<uint32, AreaTriggerTemplate> _areaTriggerTemplateStore;
+    std::unordered_map<uint32, std::vector<AreaTriggerServerPosition>> _areaTriggersByMap;
```

Comment:
This will become `std::unordered_map<ObjectGuid::LowType, AreaTriggerSpawn> _areaTriggerSpawns;` (see comments on AreaTriggerServerPosition)